### PR TITLE
Story/quad 74

### PR DESCRIPTION
**Pull request by v1kd**
Adds combined network visualizations

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 1]
```diff
@@ -56,8 +58,48 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
+		for (INetwork network: networkList) {
+
+			try {
+				List<INetworkNodeInfo> localNetworkNodeInfoList =
+						networkManager.getNetworkTopNodes(network.getNetworkId());
+				if (localNetworkNodeInfoList != null) {
+					networkNodeInfoList.addAll(localNetworkNodeInfoList);
+				}
+			} catch (QuadrigaStorageException e) {
+				logger.error("DB Error while getting network top nodes for a network with id - " +
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: don't swallow the exception. The user doesn't know what happening if you just log the exception and forget about it.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 2]
```diff
@@ -56,8 +58,48 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
 
         return networkJSon;
     }
+
+	@Override
+	public ITransformedNetwork getTransformedNetworkOfProject(String projectId)
+			throws QuadrigaStorageException {
+
+		List<INetwork> networkList;
+		ITransformedNetwork networkJSon;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: declare varaibles where they are needed

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 3]
```diff
@@ -56,8 +58,48 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
+			logger.error("DB Error while getting networks a project id", e);
+			return null;
+		}
+
+		if (networkList == null) {
+			return null;
+		}
+
+		for (INetwork network: networkList) {
+
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: remove empty line

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 4]
```diff
@@ -56,8 +58,48 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
+					networkNodeInfoList.addAll(localNetworkNodeInfoList);
+				}
+			} catch (QuadrigaStorageException e) {
+				logger.error("DB Error while getting network top nodes for a network with id - " +
+						network.getNetworkId(), e);
+			}
+
+		}
+
+		networkJSon = transformer.transformNetwork(networkNodeInfoList);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: merge with line 103

**Comment by jdamerow** [comment]
please resolve conflicts and a few small changes

**Comment by v1kd** [comment]
Refactored the code as per comments. Please review

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 5]
```diff
@@ -14,50 +16,80 @@
...
+				if (localNetworkNodeInfoList != null) {
+					networkNodeInfoList.addAll(localNetworkNodeInfoList);
+				}
+			} catch (QuadrigaStorageException e) {
+				throw new QuadrigaStorageException("Database Error while getting network top nodes for a network with ID: " +
+						network.getNetworkId(), e);
+			}
+		}
+
+		ITransformedNetwork networkJSon = transformer.transformNetwork(networkNodeInfoList);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java:92]: merge with next line

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
I'll merge this pull request. Just do the one small change with your next story.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java (hunk 5 above)
Comment (v1kd) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java:92]: I will make this change in the next story

**Pull request merged**